tgui/packages/tgui/interfaces/TEG.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author Sovexe (https://github.com/Sovexe)
+ * @license ISC
+ */
+
+import { useBackend } from '../backend';
+import { Box, Chart, LabeledList, Section, Divider } from '../components';
+import { formatPower, formatSiUnit } from '../format';
+import { Window } from '../layouts';
+
+export const TEG = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    output,
+    history,
+    hotCircStatus,
+    hotInletTemp,
+    hotOutletTemp,
+    hotInletPres,
+    hotOutletPres,
+    coldCircStatus,
+    coldInletTemp,
+    coldOutletTemp,
+    coldInletPres,
+    coldOutletPres,
+  } = data;
```

Comment:
For reference, if using the nested object thing as mentioned in the DM file, this would look like:
```
  const {
    output,
    history,
    hot: {
      circStatus: hotCircStatus = null,
      inletTemp: hotInletTemp = 0,
      outletTemp: hotOutletTemp = 0,
      inletPres: hotInletPres = 0,
      outletPres: hotOutletPres = 0,
    } = {},
    cold: {
      circStatus: coldCircStatus = null,
      inletTemp: coldInletTemp = 0,
      outletTemp: coldOutletTemp = 0,
      inletPres: coldInletPres = 0,
      outletPres: coldOutletPres = 0,
    } = {},
  } = data;
```